plugins/modules/backup_restore_job_info.py

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = r"""
+---
+module: backup_restore_job_info
+version_added: 6.0.0
+short_description: List information about backup restore jobs.
+description:
+    - List detailed information about AWS Backup restore jobs initiated to restore a saved resource.
+author:
+    - Mandar Vijay Kulkarni (@mandar242)
+options:
+  account_id:
+    description:
+      - The account ID to list the restore jobs from.
+    required: false
+    type: str
+  status:
+    description:
+      - Status of restore jobs to filter the result based on job status.
+    required: false
+    choices: ['PENDING', 'RUNNING', 'COMPLETED', 'ABORTED', 'FAILED']
+    type: str
+  created_before:
```

Comment:
please provide an example or a definition of the date format (same for other date parameters)
Even though it is part of the example section, could be worth adding it here